### PR TITLE
[BugFix] Skip the temporary directory created by spark when list hive directory recursively

### LINKED ISSUE #42060: [Bug] Querying the Hive table failed because it didn't skip the “_temporary” directory
I find StarRocks 2.5.x will skip _tempory dir created by spark（In Spark, a workspace directory named _temporary/ is created under the directory where the table is located to store the intermediate work, and all apps share this workspace） while StarRocks 3.x dont when query hive table

### Steps to reproduce the behavior (Required)
1. Create a temporary directory in specific in a hive table  named tmp.test
```
hdfs://data/tmp/vom/test/dt=2023-10-11/_temporary/0/_temporary/...files
```
2. Refresh external partition
```
refresh external table hive_catalog.tmp.test partition("dt=2023-10-11");
```
3. watch Fe behavior using arthas
```
watch com.starrocks.connector.CachingRemoteFileIO  loadRemoteFiles "{returnObj}" -s -x 2 
``` 
4. query hive table failed
```
select * from hive_catalog.tmp.test where dt = '2023-10-11'  failed , error msg:
ERROR 1064 (HY000): Parquet file size is 4 bytes, smaller than the minimum parquet file footer (8 bytes)
```

### Expected behavior (Required)
skip _temporary and query success
### Real behavior (Required)
dont skip  files from _temporary dir & query failed
```
[arthas@530181]$ watch com.starrocks.connector.CachingRemoteFileIO  loadRemoteFiles "{returnObj}" -s -x 2 "params[0].path.contains('test')"
Press Q or Ctrl+C to abort.
Affect(class count: 1 , method count: 1) cost in 160 ms, listenerId: 12
method=com.starrocks.connector.CachingRemoteFileIO.loadRemoteFiles location=AtExit
ts=2024-03-04 15:54:55; [cost=11.725592ms] result=@ArrayList[
    @ArrayList[
        @RemoteFileDesc[RemoteFileDesc{fileName='_temporary/0/_temporary/attempt_20240123104122_0008_m_000000_1470/part-00000-79dfc67c-c157-4f3e-bc3e-c410b6d167d9-c000.snappy.parquet', compression='', length=4, modificationTime=1705977697753, blockDescs=[RemoteFileBlockDesc{offset=0, length=4, replicaHostIds=[140], 133], 443], diskIds=[-1], remoteFileIO=com.starrocks.connector.hive.HiveRemoteFileIO@9b4f9a0}], splittable=false, textFileFormatDesc=null, hudiDeltaLogs=[], icebergScanTasks=[], paimonSplitsInfo=null}],
        @RemoteFileDesc[RemoteFileDesc{fileName='_temporary/0/_temporary/attempt_20240123104122_0008_m_000003_1473/part-00003-79dfc67c-c157-4f3e-bc3e-c410b6d167d9-c000.snappy.parquet', compression='', length=2193, modificationTime=1705977697748, blockDescs=[RemoteFileBlockDesc{offset=0, length=2193, replicaHostIds=[709], 339], 226], diskIds=[-1], remoteFileIO=com.starrocks.connector.hive.HiveRemoteFileIO@9b4f9a0}], splittable=false, textFileFormatDesc=null, hudiDeltaLogs=[], icebergScanTasks=[], paimonSplitsInfo=null}],
    ],
]
```

### StarRocks version (Required)
- 3.1.8
- 3.2.3

### PR DESCRIPTION
**Pull request by haiboself**
## Why I'm doing:
Spark job/sql will created temporary named _temporary(see: https://stackoverflow.com/questions/46882683/spark-temporary-creation-reason) to storing  intermediate work for ensuring consistency of the final result.  StarRocks does not skip the _temporary directory and its files when querying a Hive table, which can cause the query to fail or yield incorrect results.

## What I'm doing:
Skip directory named _temporary when recursively listing Hive directories

Fixes #42060

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [x] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ private boolean isValidDirectory(FileStatus fileStatus) {
             return false;
         }
         String dirName = fileStatus.getPath().getName();
-        return !(dirName.startsWith("."));
+        return !(dirName.startsWith(".") || dirName.startsWith("_temporary"));
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java:191]: ignore files that satisfies `dirName.startsWith("_")`, aligning with `isValidDataFile`

Comment (haiboself) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java]: get

**Review by letian-jiang — APPROVED** [review]
LGTM

**Review by stephen-shelby — APPROVED** [review]

**Comment by haiboself** [comment]
> @haiboself Please sign the CLA, and then the commit can be accepted.

done, thanks~

**Review by letian-jiang — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**